pandas/_libs/tslibs/conversion.pyx

```diff
@@ -595,7 +595,7 @@ cdef check_overflows(_TSObject obj, NPY_DATETIMEUNIT reso=NPY_FR_ns):
 # ----------------------------------------------------------------------
 # Localization
 
-cdef void _localize_tso(_TSObject obj, tzinfo tz, NPY_DATETIMEUNIT reso):
+cdef void _localize_tso(_TSObject obj, tzinfo tz, NPY_DATETIMEUNIT reso) noexcept:
```

Comment:
Is this really `noexcept`? I think the `Localizer` constructor can raise

Reply:
In Cython 0.29.X the default exception handling is `noexcept` hence the exception was not propagated even before this PR. If an exception should be propagated, one should consider changing return type from `void` to `int` to avoid checking exception after every function call.